### ISSUE TITLE
Is it possible to add a backlink to current org-mode file in the generated Anki card?

### ISSUE THREAD

**Issue by panmengguan** [eyeinsky/org-anki#18]
I think this kind of backlink would be very helpful. Is there any idea? My elisp is poor, however, if you have any idea but do not have time to implement, I may have a try and creat a pull request.

Generating a backlink to the org-exported html instead may also be helpful, but I have tried and found that anki does not support link to local html files for security concern. So I don't know how to solve this.

Thank you very much for this package. I think it is more suitable in my working flow than the existing anki-editor package.

**Comment by eyeinsky** [comment]
May I ask why this would be helpful to you?

**Comment by mgpanpan** [comment]
The Anki cards generated by the marginnote app (https://www.marginnote.com/) have this type of backlink. When reviewing the anki cards generated by it, a backlink is provided to trace the source and context of this card. So I hope the cards generated by emacs org-mode could have similar function.

Just now, I have an idea that this may be implemented based on org-protocol.

**Comment by panmengguan** [comment]
> May I ask why this would be helpful to you?

The Anki cards generated by the marginnote app (https://www.marginnote.com/) have this type of backlink. When reviewing the anki cards generated by it, a backlink is provided to trace the source and context of this card. So I hope the cards generated by emacs org-mode could have similar function.

Just now, I have an idea that this may be implemented based on org-protocol.

**Comment by eyeinsky** [comment]
@mgpanpan Would the back-link you have in mind be something like this:

Given the org file in /home/username/org/my-cards.org:
```org
* My question

  my answer
```

The note back (body) would be:
```html
<p>my answer</p>
<p>Source: <a href="/home/username/org/my-cards.org">/home/username/org/my-cards.org</a></p>
```

**Comment by panmengguan** [comment]
> @mgpanpan Would the back-link you have in mind be something like this:
> 
> Given the org file in /home/username/org/my-cards.org:
> 
> ```
> * My question
> 
>   my answer
> ```
> 
> The note back (body) would be:
> 
> ```
> <p>my answer</p>
> <p>Source: <a href="/home/username/org/my-cards.org">/home/username/org/my-cards.org</a></p>
> ```

Yes, that would be nice. I think it could be better if the link is : `<a href="/home/username/org/my-cards.org">title of my-cards</a>`.

In fact, I have also modified this package to add a backlink with org-protocol support for my private usage this period. E.g.:
```
<a href="org-protocol://roam-file?file=/home/username/org/my-cards.org"> title of my-cards</a>
```

This can make emacs to open the org-mode file. However, it depends on the org-protocol added by org-roam. I don't know how to do it generally.

**Comment by eyeinsky** [comment]
Note that title of the card is the title of the Anki note, too (so you don't necessarily need to add it to the link in bottom since you already know it from the card's title).

But otherwise I'm thinking this: add some setting called `ANKI_FOOTER` (or something), where one could specify the syntax for the footer themselves. So at the top of the .org file you write something like:
```
ANKI_FOOTER: <a href="org-protocol://$path">$front</a>
```
and get that rendered at the bottom of every note. This way if people have other ideas for what they want at the footer, they could modify the template.

**Comment by panmengguan** [comment]
Yes, that is a good idea. Maybe this parameter could also be configured globally for all files.

**Comment by hwiorn** [comment]
I made [a plugin for org backlink](https://github.com/hwiorn/anki-open-org-note) for my own workflow need. This plugin doesn't make any extra fields for backlink. Just searching for `ANKI_NOTE_ID` in org directory and it execute `emacsclient` to show org note. I can't say it is well-made or fit everyone. If you need a backlink feature, give it a try.

**Comment by eyeinsky** [comment]
There is now a PR attempting to solve this issue https://github.com/eyeinsky/org-anki/pull/58.

**Comment by panmengguan** [comment]
> There is now a PR attempting to solve this issue #58.

Thanks!

**Comment by eyeinsky** [comment]
Merged PR as version v3.0.0.
